doc/usage/extensions/autosectionlabel.rst

```diff
@@ -39,6 +39,13 @@ Configuration
    avoiding ambiguity when the same section heading appears in different
    documents.
 
+   When ``index.rst`` doesn't reside at your docs root directory, the prefix
+   has to contain any directories relative to it.
+
+   Example: if ``index.rst`` lives at ``example/hello/world/index.rst``, use
+   ``:ref:`example/hello/world/index:Introduction```. (Carefull: no leading
+   ``/`` as necessary with ``:doc:``!)
```

Comment:
`:doc:` role can accept docname in relative and absolute form. It is a special case of "document name". I think it is not good example...